sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -466,6 +457,41 @@ public final class StorageBlobClient: PipelineClient {
         url?.appendPathComponent(blob)
         return url
     }
+
+    /// Generates an appropriate download URL for a blob based on specific download options.
+    /// - Parameters:
+    ///   - blob: The name of the blob.
+    ///   - container: The name of the container.
+    ///   - options: A `DownloadBlobOptions` object to control the download operation.
+    /// - Throws: `AzureError.fileSystem` when cache directory cannot be found.
+    /// - Returns: Destination `URL` for the blob download.
+    public func downloadUrl(
+        forBlob blob: String,
+        inContainer container: String,
+        withOptions options: DownloadBlobOptions? = nil
+    ) throws -> URL {
```

Comment:
This is definitely worthwhile to keep in the client. This is the logic that I was previous created a download specifically for, and it doesn't make sense to make the client copy this snippet.

Reply:
Moved to PathHelpers.